ibmdbpy/frame.py

```diff
@@ -2316,3 +2323,106 @@ def check_numeric_columns(idaobject):
         check_numeric_columns(self)
         if isinstance(other, ibmdbpy.IdaSeries)|isinstance(other, IdaDataFrame):
             check_numeric_columns(other)
+
+###############################################################################
+### New functionalities for IdaDataFrames -- Experimental
+###############################################################################
+    
+    def delete_na(self, columns, logic = "any", inplace = False):
+        """
+        
+        Parameters
+        -------
+        columns: list of eligible column names
+        logic: str, optional, "any" by default. If logic is set to "any" then all rows which contain a NaN value in any 
+        (id est at least one) of the cited columns is deleted, this is a union condition; 
+        if logic is set to "all", then only rows containing null values in all the fields are deleted, this is an intersection condition.         
+        inplace: bool, optional, False by default. If True, then the underlying table is physically modified,
+        if False, then the original objects remain unaffected, a copy of the IdaDataFrame is made and modified. 
+        
+        Returns
+        -------
+        If inplace is True, the original table is modified and the IdaDataFrame will be modified accordingly. No return.
+        If inplace is False, a new IdaDataFrame is returned, the original table is not affected. This new IdaDataFrame points to a 
+        new table which has been created according to the user defined criterions on columns.
+        
+        Examples
+        -------
+        idadf.delete_na(["COL_1", "COL_2"], inplace = True)
+        # each row of the original table is physically deleted if there is a NULL value in one of the listed columns
+        new_idadf = idadf.delete_na(["COL_1", "COL_2"], union = False)
+        # a new IdaDataFrame is created, rows will be selected only if all the listed columns have a NULL value.
+        """
+        
+        # Check if list columns is not empty
+        if len(columns)<1:
+            raise IdaDataFrameError("You must specify the columns as a list of eligible names")
+        # Check if column names are eligible
+        for column_name in columns:
+            if not isinstance(column_name, six.string_types):
+                raise TypeError("%s is not of string type"%(column_name))
+            if column_name not in self.columns:
+                raise ValueError("%s refers to a columns that doesn't exists in self"%(column_name))
+        
+        idadb = self._idadb
+        
+        if inplace == True:
+            # Ask for confirmation
+            print("The table %s will be physically modified." %self.tablename)
+            answer = input('Do you want to proceed? [y/n]')
```

Comment:
Please do not add `input` calls to ibmdbpy. It's a library. As such, while it can be used in jupyter notebooks, nothing stops users from using it as a library in code that runs non-interactively (for example runs over night without human supervision). In those cases, noone is around to answer the question.

Reply:
Also, it's nearly impossible to write automated test cases for code that uses `input`.

Reply:
![image](https://user-images.githubusercontent.com/55744208/74664721-8964f900-519e-11ea-934c-9e917855fcf2.png)
I understand it is not practical to use `input`here. However how do you suggest tackling this issue otherwise? Let me explain:

In the example above, query should be:
CREATE VIEW "TEST_VIEW" AS 
(SELECT IDA1."USER_ID" AS "INDEXERIDA1",IDA2."ID" AS "INDEXERIDA2",DB2GSE.ST_CONTAINS(DB2GSE.ST_BUFFER(IDA1.USER_LOC,50),IDA2.GEOM) AS "RESULT" 
FROM 
(SELECT "USER_ID","USER_LOC","MUSEUM_TYPE","MAX_DIST",DB2GSE.ST_BUFFER(USER_LOC,50) AS "DEFAULT_BUFFER" FROM BLUADMIN.GEO_TOURISTS) AS IDA1, 
BLUADMIN.GEO_MUSEUMS AS IDA2);
But due to the structure of the code, you cannot change “IDA1.DB2GSE_FUNCTION(COL, ARG)” into “DB2GSE_FUNCTION(IDA1.COL,ARG)” automatically because it is derived from other queries. So the fix relied on the user input: if an intricated function pattern is detected in the query, user input is asked to avoid confusion. An explanation is prompted and the user is asked for the geometry column name. 
Do you see another way to resolve this issue?